five-minute-serverless-backend/function.js

```diff
@@ -4,10 +4,11 @@ const CORS = (handler) =>
       statusCode: 200,
       headers: {
         'Access-Control-Allow-Origin': '*',
-        'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
       },
     };
-    if (context.request.method !== 'OPTIONS') {
+    if (context.request.method === 'OPTIONS') {
+      response.headers['Access-Control-Allow-Headers'] = context.request.headers['access-control-request-headers'];
```

Comment:
Just to make this future proof, can you also check that `context.request.headers['access-control-request-headers']` is defined? (It will definitely be defined in this tutorial, but a usage for e.g. an endpoint for the DELETE method would likely need handling `access-control-request-method` but not access-control-request-headers.